controllers/openstackmachine_controller.go

```diff
@@ -530,6 +528,22 @@ func (r *OpenStackMachineReconciler) reconcileDeleteFloatingAddressFromPool(scop
 	return r.Client.Update(context.Background(), claim)
 }
 
+// reconcileDeleteMachineServer reconciles the deletion of the OpenStackServer.
+// It returns nil if the OpenStackServer is deleted or not found or an error if there is any.
+func (r *OpenStackMachineReconciler) reconcileDeleteMachineServer(ctx context.Context, scope *scope.WithLogger, openStackMachine *infrav1.OpenStackMachine) error {
+	scope.Logger().Info("Reconciling Machine delete server")
+	server := &infrav1alpha1.OpenStackServer{}
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: openStackMachine.Namespace, Name: openStackMachine.Name}, server)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+
```

Comment:
If memory works, I added this to avoid misleading logs where an object that doesn't exist would tried to be deleted but couldn't be found.
I'll add a commit that remove that block but I'll re-check what's in the logs when that happens.

Reply:
Done via: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2067/commits/e415725d5adcef6a0a8f0493da63c52df7540a54
 Will be squashed once reviewed again.